### DISCUSSION TITLE
highlight.js syntax highlighting only works locally?

### DISCUSSION THREAD

**Discussion by monicathieu** [thuliteio/doks#1066] [Q&A] [answered]
Hi, Doks is amazing first off! Thank you Henk and everyone who's worked on it!

My issue is as in the title: I believe I've done everything to enable highlight.js syntax highlighting, but it only seems to work locally.

My website repo: https://github.com/monicathieu/spelman-wistem-psych
My hosted website, through Netlify: https://spelman-wistem-psych.netlify.app/

I turned on the following settings to use highlight.js with my desired theme per the Doks docs:

```toml
# in config/_default/params.toml
[options]
  ...
  highLight = true
``` 

```css
/* in assets/scss/app.scss */
@import "highlight.js/scss/base16/unikitty-light";
```

The website is for an R data science summer workshop, so I added R as a language to highlight:

```js
// in assets/js/highlight.js
// in the imports section
import r from 'highlight.js/lib/languages/r';

// in the language registration section
hljs.registerLanguage('r', r);
```

When I run the website locally using `npm run start`, the highlighting looks nice:

![Screen Shot 2023-06-16 at 4 12 38 PM](https://github.com/h-enk/doks/assets/9875687/6d87fadc-bb6b-45ac-a8d3-1b25290292ce)

But when the website builds on Netlify, my theme appears not to be working! [(relevant blog post here)](https://spelman-wistem-psych.netlify.app/blog/class-notes-june-15-2023/)

![Screen Shot 2023-06-16 at 4 12 50 PM](https://github.com/h-enk/doks/assets/9875687/7f16ad2a-6cce-4f9e-8839-f745aea54d19)

I'm not 100% sure what default theme is being used, so I haven't been able to diagnose whether highlight.js is working ok on Netlify and it's just failing to import my theme, or whether maybe this is actually Chroma highlighting and highlight.js isn't working at all.

Can you assist?

**Comment by h-enk** [comment, marked as answer]
Hi Monica, nice to hear that you like Doks!

It's just a small setting. The highlight.js theme's CSS gets removed by PurgeCSS when building the site for production.

So, you'll need to explicitly whitelist the `unikitty-light.scss` file (and comment out the default `_syntax.scss`) in [postcss.config.js](https://github.com/monicathieu/spelman-wistem-psych/blob/main/config/postcss.config.js#L32) like so (line 32..33):

```js
[..]
        ...whitelister([
          './node_modules/@hyas/doks/assets/scss/common/_variables.scss',
          './node_modules/@hyas/doks/assets/scss/components/_alerts.scss',
          './node_modules/@hyas/doks/assets/scss/components/_buttons.scss',
          './node_modules/@hyas/doks/assets/scss/components/_code.scss',
          // './node_modules/@hyas/doks/assets/scss/components/_syntax.scss',
          './node_modules/highlight.js/scss/base16/unikitty-light.scss',
          './node_modules/@hyas/doks/assets/scss/components/_search.scss',
          './node_modules/@hyas/doks/assets/scss/common/_dark.scss',
          './node_modules/katex/dist/katex.css',
        ]),
[..]
```

**Reply by monicathieu** [reply]
Amazing, thank you--made those changes to the whitelister list and the Netlify site now uses my highlight.js theme!

Would it be helpful if I submitted a PR to the docs adding this to the instructions for changing the highlight.js theme?

**Reply by h-enk** [reply]
Super! 

> Would it be helpful if I submitted a PR to the docs adding this to the instructions for changing the highlight.js theme?

Thanks, but don't bother. I'm working on the new documentation site for Doks 1.0 (available soon). I will include this!

**Comment by Yvand** [comment]
I have exactly the same symptoms using `"@hyas/doks-core": "^1.0.7"`.
I've been trying to use the same steps [as your answer](https://github.com/h-enk/doks/discussions/1066#discussioncomment-6203146) to fix it on the live site, but somehow it has no effect.

Site page showing the highlight issue:  https://entracp.yvand.net/docs/usage/installation/
GitHub project: https://github.com/Yvand/Hugo-EntraCP
My PR to attempt to fix it: https://github.com/Yvand/Hugo-EntraCP/pull/4

I'm a bit out of ideas what to do. Can you please help?

**Reply by h-enk** [reply]
Use the included styles by setting `noClasses = true` in `config/_default/markup.toml` — see also: https://github.com/Yvand/Hugo-EntraCP/blob/main/config/_default/markup.toml#L25

**Reply by Yvand** [reply]
It works ! I would have never found this myself, thank you !
